### PR TITLE
Replace gulp with prettier and add isort

### PR DESCRIPTION
**Pull request by LJBabbage**
### PR Context
Dependabot highlighted an issue with set-value and glob-parent, which now conflict with gulp 4.0.2 (latest version which has been updated since 2019) This PR aligns runner and validator and replaces gulp with prettier, I have also added isort as well.

N.B I have updated pull_request.xml to have just one linting step, naturally this means that it won't pass the old ones (lint-python and lint-json) but will pass the new one. I will update the repo before it gets merged as naturally there are other PR in flight that won't have the new yaml

### Testing
Make sure format and linting work by messing around with some JSON formatting and checking they are identified(lint) or fixed (format)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,36 +1,71 @@
 {
```
Comment (MebinAbraham) [inline .eslintrc.json:1]: Was this updated just to be consistent with runner?. If so, the runner specific rules should be removed, there are config that only apply to runner such as specifics relating to webdriver and chai etc.

File: .eslintrc.json [hunk 2]
```diff
@@ -1,36 +1,71 @@
...
+    "plugin:import/recommended"
+  ],
+  "globals": {
+    "$": true,
+    "$$": true,
+    "browser": true,
+    "expect": true
+  },
+  "parserOptions": {
+    "ecmaVersion": 11,
```
Comment (MebinAbraham) [inline .eslintrc.json:19]: We should upgrade this to the latest and set it to `12` or `2021` for clarity.

File: package.json [hunk 3]
```diff
@@ -9,7 +9,12 @@
   },
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "lint": "gulp lint"
+    "lint": "yarn lint:schemas && yarn lint:test-schemas",
+    "lint:schemas": "prettier --check \"schemas/**/*.json\" && eslint \"schemas/**/*.json\"",
+    "lint:test-schemas": "prettier --check \"tests/schemas/*/*.json\" && eslint \"tests/schemas/**/*.json\"",
+    "format": "yarn format:schemas && yarn format:test-schemas",
+    "format:schemas": "prettier \"schemas/*/*.json\" --write && eslint --fix \"schemas/*/*.json\"",
+    "format:test-schemas": "prettier \"tests/schemas/*/*.json\" --write && eslint --fix \"tests/schemas/*/*.json\""
```
Comment (MebinAbraham) [inline package.json:17]: The globs need to be updated to supports nested directory that aren't just at the top level. At the moment, not all files are picked up. 

Also, we seem to be not linting and formatting the `ajv` folder, but we should be. `AJV` folder fails lint atm and should be fixed.
```suggestion
    "lint": "yarn lint:schemas && yarn lint:test-schemas && yarn lint:ajv",
    "lint:schemas": "prettier --check \"schemas/**/*.json\" && eslint \"schemas/**/*.json\"",
    "lint:test-schemas": "prettier --check \"tests/schemas/**/*.json\" && eslint \"tests/schemas/**/*.json\"",
    "lint:ajv": "prettier \"ajv/**/*.js\" && eslint \"ajv/**/*.js\"",
    "format": "yarn format:schemas && yarn format:test-schemas && yarn format:ajv",
    "format:schemas": "prettier \"schemas/**/*.json\" --write && eslint --fix \"schemas/**/*.json\"",
    "format:test-schemas": "prettier \"tests/schemas/**/*.json\" --write && eslint --fix \"tests/schemas/**/*.json\"",
    "format:ajv": "prettier \"ajv/**/*.js\" --write && eslint --fix \"ajv/**/*.js\""
```

**Review by MebinAbraham — COMMENTED** [review]
👍 for add `isort`.

- The `gulpfile.js` should now be removed.
- We should also lint/format the `ajv` folder, it seemed to have been missed before.

File: .eslintrc.json (hunk 1 above)
Comment (LJBabbage) [reply .eslintrc.json:1]: Yeah, I just lifted, and didn't update, same with version, although I did in packages.json

File: package.json (hunk 3 above)
Comment (LJBabbage) [reply package.json]: Well that broke :laughing:

File: .eslintrc.json [hunk 4]
```diff
@@ -1,36 +1,68 @@
 {
-    "env": {
-        "browser": true,
-        "es6": true
-    },
-    "plugins": [
-        "security"
```
Comment (MebinAbraham) [inline .eslintrc.json:7]: This should be kept, along with:
```json
"extends": [
        "plugin:security/recommended"
]
```

File: .eslintrc.json [hunk 5]
```diff
@@ -1,36 +1,68 @@
...
+  "extends": [
+    "standard",
+    "eslint:recommended",
+    "plugin:import/recommended"
+  ],
+  "globals": {
+    "$": true,
+    "$$": true,
+    "browser": true,
+    "expect": true
```
Comment (MebinAbraham) [inline .eslintrc.json:15]: These are redundant?

File: .eslintrc.json (hunk 4 above)
Comment (LJBabbage) [reply .eslintrc.json:7]: They must not have been in runner, as I didn't explicitly removed them

**Review by MebinAbraham — APPROVED** [review]

**Review by petechd — APPROVED** [review]

**Pull request merged**